### PR TITLE
Handle when user hits Cancel on OAuth prompt (#187)

### LINKED ISSUE #187: Handle cleanly when user hits "Cancel" on Canvas authorize prompt
As the OAuth flow currently exists, if the user hits "Cancel" when they reach Canvas, the application errors out. Because the returned URL query has only an `error` key, it's not considered a "bad" query, and we end up making an invalid call to request a Canvas token (resulting in a `500` error on the request to `canvas/returnFromOAuth`).

Work on issue #101 is somewhat adjacent to this. We either need to show an error page, or redirect back to the application which then restarts the Canvas OAuth flow.

### PR DESCRIPTION
**Pull request by ssciolla**
The PR aims to resolve #187.

### CODE REVIEW

File: ccm_web/server/src/canvas/canvas.controller.ts [hunk 1]
```diff
@@ -47,15 +47,24 @@ export class CanvasController {
...
+        logger.debug('User rejected Canvas OAuth; sending them back to application root...')
+        res.redirect('/')
+        return
+      }
+
+      logger.error(`Canvas OAuth failed due to ${query.error}: ${String(query.error_description)}`)
+      const message = query.error_description !== undefined
+        ? query.error_description
+        : 'Canvas OAuth error was sent without a description.'
+      throw new InternalServerErrorException(message)
```
Comment (pushyamig) [inline ccm_web/server/src/canvas/canvas.controller.ts:62]: Does this code ever will ever hit? what are other scenarios to apart from access_denied to reach L58

Comment (ssciolla) [reply ccm_web/server/src/canvas/canvas.controller.ts:62]: I think you added this before in PR #130 to handle a scope error. So yes? I think even still, it's good to have a general fallback for scenarios you haven't anticipated.

Comment (pushyamig) [reply ccm_web/server/src/canvas/canvas.controller.ts:62]: Yeah.. I remember that. Agreed on fallback. 

But when I coded that it was sending the error message like ` Canvas OAuth failed due to invalid_scope: A requested scope is invalid, unknown, malformed, or exceeds the scope granted by the resource owner. The following scopes were requested, but not granted: url:POST|/api/v1/accounts/:account_id/account_notifications`
Not anymore. The UI just be as below
Screen Shot 2021-09-24 at 1 55 03 PM

This is a separate issue.

**Review by pushyamig — APPROVED** [review]
Looks Good! Works as intended to.

File: ccm_web/server/src/canvas/canvas.controller.ts (hunk 1 above)
Comment (ssciolla) [reply ccm_web/server/src/canvas/canvas.controller.ts:62]: I don't remember that, but I'd have to look back to see.

**Pull request merged**